packages/builders/src/components/button/mixins/URLOrCustomIdButtonMixin.ts

```diff
@@ -0,0 +1,44 @@
+import type { APIButtonComponent, APIButtonComponentWithSKUId, APIMessageComponentEmoji } from 'discord-api-types/v10';
+
+export interface URLOrCustomIdButtonData
+	extends Pick<Exclude<APIButtonComponent, APIButtonComponentWithSKUId>, 'emoji' | 'label'> {}
+
+export class URLOrCustomIdButtonMixin {
```

Comment:
I think a name like `EmojiOrLabelButtonMixin` or something along those lines fits this better